jdbc-yugabytedb/src/main/java/org/postgresql/ds/common/BaseDataSource.java

```diff
@@ -1183,7 +1192,9 @@ public void setProperty(PGProperty property, String value) {
         break;
       case PG_PORT:
         try {
-          portNumber = Integer.parseInt(value);
+          if (portNumber == 0) {
+            portNumber = Integer.parseInt(value);
+          }
```

Comment:
what is the overriding rule here? For the other cases it looks like we're not checking the previous value but just overriding whatever was there before. Why are we differing for port (i.e. only setting it if it is `0`/unset previously)?

Reply:
There is a separate addPort method in the datasource. We are giving priority to that if set explicitly.